Sources/StreamChatSwiftUI/CommonViews/AlertBannerViewModifier.swift

```diff
@@ -0,0 +1,89 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import SwiftUI
+
+extension View {
+    /// Presents an alert banner with a title at the top of the view.
+    ///
+    /// - Parameters:
+    ///   - title: A text string used as the title of the alert banner.
+    ///   - isPresented: A binding to a Boolean value that determines whether to present the alert banner.
+    ///   - action: An action which is added to the view through refreshable view modifier.
+    ///   - duration: The amount if time after which the banner is dismissed automatically.
+    func alertBanner(
+        _ title: String = L10n.Alert.Error.title,
+        isPresented: Binding<Bool>,
+        action: (() -> Void)? = nil,
+        duration: TimeInterval = 3
+    ) -> some View {
+        modifier(
+            AlertBannerViewModifier(
+                title: title,
+                isPresented: isPresented,
+                action: action,
+                duration: duration
+            )
+        )
+    }
+}
+
+private struct AlertBannerViewModifier: ViewModifier {
+    @Injected(\.colors) private var colors
+    let title: String
+    @Binding var isPresented: Bool
+    let action: (() -> Void)?
+    let duration: TimeInterval
+    @State private var timer: Timer?
+    
+    func body(content: Content) -> some View {
+        VStack(spacing: 0) {
+            VStack {
+                if isPresented {
+                    Text(title)
+                        .font(.body)
+                        .foregroundColor(Color(colors.staticColorText))
+                        .padding(.init(top: 4, leading: 16, bottom: 4, trailing: 16))
+                        .frame(maxWidth: .infinity)
+                        .background(Color(colors.textLowEmphasis))
+                        .transition(.move(edge: .top))
+                }
+            }
+            .frame(maxWidth: .infinity)
+            .contentShape(Rectangle())
+            .clipped()
+            content
+        }
+        .alertBannerRefreshable(action: action)
+        .animation(.easeIn, value: isPresented)
+        .onChange(of: isPresented) { newValue in
+            guard newValue else { return }
+            timer?.invalidate()
+            timer = Timer.scheduledTimer(withTimeInterval: duration, repeats: false) { _ in
+                isPresented = false
+            }
+        }
+    }
+}
+
+// MARK: - Refreshable Compatibility
+
+private struct AlertBannerActionViewModifier: ViewModifier {
+    let action: (() -> Void)?
+    
+    func body(content: Content) -> some View {
+        if #available(iOS 15.0, *), let action {
+            content
+                .refreshable { action() }
```

Comment:
this one got me thinking a bit - probably the refreshable part can be independent of the banner, so we can easily add it for e.g. for reloading channels / messaging. Outside the scope of this one, but worth revisiting it we still need this here.

Reply:
It was added because otherwise there is nothing user can do when an error banner is displayed. I'll add a ticket about revisiting this and other views.

Reply:
@martinmitrevski what do you think, should I pull out the refreshable part for now and we will revisit it in [880](https://github.com/GetStream/ios-issues-tracking/issues/880). I am OK to do refreshable part separately.

Reply:
let's keep it here and extend it in the other ticket.